volatility3/framework/plugins/windows/psxview.py

```diff
@@ -0,0 +1,279 @@
+
+    @classmethod
+    def get_requirements(cls):
+        return [
+            requirements.ModuleRequirement(
+                name="kernel",
+                description="Windows kernel",
+                architectures=["Intel32", "Intel64"],
+            ),
+            requirements.VersionRequirement(
+                name="info", component=info.Info, version=(1, 0, 0)
+            ),
+            requirements.VersionRequirement(
+                name="pslist", component=pslist.PsList, version=(2, 0, 0)
+            ),
+            requirements.VersionRequirement(
+                name="psscan", component=psscan.PsScan, version=(1, 0, 0)
+            ),
+            requirements.VersionRequirement(
+                name="thrdscan", component=thrdscan.ThrdScan, version=(1, 0, 0)
+            ),
+            requirements.VersionRequirement(
+                name="handles", component=handles.Handles, version=(1, 0, 0)
+            ),
+            requirements.VersionRequirement(
+                name="sessions", component=sessions.Sessions, version=(0, 0, 0)
+            ),
+            requirements.BooleanRequirement(
+                name="identify-expected",
+                description='In the plugin\'s output, replace false with \
+                                                normal where false is the expected result for a Windows machine running normally. \
+                                                Keep in mind that this plugin uses simple checks to identify "normal" behavior, \
+                                                so you may want to double-check the legitimacy of these processes yourself.',
+                optional=True,
+            ),
+            requirements.BooleanRequirement(
+                name="physical-offsets",
+                description="List processes with phyiscall offsets instead of virtual offsets.",
+                optional=True,
+            ),
+        ]
+
+    def proc_name_to_string(self, proc):
+        return proc.ImageFileName.cast(
+            "string", max_length=proc.ImageFileName.vol.count, errors="replace"
+        )
+
+    def is_ascii(self, str):
+        return str.split(".")[0].isalnum()
+
+    def filter_garbage_procs(self, proc_list):
+        return [
+            p
+            for p in proc_list
+            if p.is_valid() and self.is_ascii(self.proc_name_to_string(p))
+        ]
+
+    def translate_offset(self, offset):
+        if self.config["physical-offsets"]:
+            return offset
+
+        kernel = self.context.modules[self.config["kernel"]]
+        layer_name = kernel.layer_name
+
+        try:
+            offset = list(
+                self.context.layers[layer_name].mapping(offset=offset, length=0)
+            )[0][2]
+        except:
+            # already have physical address
+            pass
+
+        return offset
+
+    def proc_list_to_dict(self, tasks):
+        return {self.translate_offset(proc.vol.offset): proc for proc in tasks}
```

Comment:
Wow, that's a new one on me, I never tried... dictionary comprehensions?  I've always made a list of tuples and then called dict on it (so `dict([(self.translate_offset(proc.vol.offset), proc) for proc in tasks])`.  This seems to work, but it did take me a minute to read it because it's so unusual...  5:)

Reply:
It's actually my first time using a dictionary comprehension. Hard to decide when the tradeoff of readability vs conciseness is worth it though